cartoframes/assets/src/layers.js

```diff
@@ -17,10 +17,10 @@ export function initMapLayer(layer, layerIndex, numLayers, hasLegends, map, mapI
     throw e;
   }
 
-  setLayerLegend(layer, mapLayerIndex, mapLayer, mapIndex, hasLegends);
-  setLayerWidgets(map, layer, mapLayer, mapLayerIndex, mapSource);
 
   mapLayer.addTo(map);
+  setLayerLegend(layer, mapLayerIndex, mapLayer, mapIndex, hasLegends);
+  setLayerWidgets(map, layer, mapLayer, mapLayerIndex, mapSource);
```

Comment:
Was the previous order causing something not ready in the Layer for the Legend & Widgets? 
If that is the case maybe it could be better to rely on 'loaded' event...

Reply:
I tried that but there's no difference, actually. I tested it and leave it this way, but there's no difference.